src/ActionMenu.tsx

```diff
@@ -45,14 +46,23 @@ const ActionMenuBase = ({
         role: 'menuitem',
         onAction: (props, event) => {
           const actionCallback = itemOnAction ?? onAction
-          actionCallback?.(props as ItemProps, event)
+          pendingActionRef.current = () => actionCallback?.(props as ItemProps, event)
           onClose()
         }
       })
     },
     [onAction, onClose, renderItem]
   )
 
+  useEffect(() => {
+    // Wait until menu has re-rendered in a closed state before triggering action.
+    // This is needed in scenarios where the action will move focus, which would otherwise be captured by focus trap
+    if (!open && pendingActionRef.current) {
+      pendingActionRef.current()
+      pendingActionRef.current = undefined
+    }
+  }, [open])
```

Comment:
To clarify: `open` is sufficient as a dependency? (i.e. we don’t need both `open` and `pendingActionRef.current`)?

Reply:
`open` should be enough here because we only want to run this effect when `open` goes from `true` to `false`, and  `pendingActionRef.current` won't ever trigger a re-render when set